packages/web/server/queries/complex/assets/price.ts

```diff
@@ -40,22 +41,50 @@ async function getCoingeckoCoin({ denom }: { denom: string }) {
   });
 }
 
+/** Used with `DataLoader` to make batched calls to CoinGecko.
+ *  This allows us to provide IDs in a batch to CoinGecko, which is more efficient than making individual calls. */
+async function batchFetchCoingeckoPrices(
+  coinGeckoIds: readonly string[],
+  currency: CoingeckoVsCurrencies
+) {
+  const pricesObject = await querySimplePrice(coinGeckoIds as string[], [
+    currency,
+  ]);
+  return coinGeckoIds.map(
+    (key) =>
+      pricesObject[key][currency] ??
+      new Error(`No CoinGecko price result for ${key} and ${currency}`)
+  );
+}
 async function getCoingeckoPrice({
   coingeckoId,
   currency,
 }: {
   coingeckoId: string;
   currency: CoingeckoVsCurrencies;
 }) {
+  /** Create a loader per given currency. */
+  const currencyBatchLoader = await cachified({
+    cache: pricesCache,
+    key: `prices-batch-loader-${currency}`,
+    getFreshValue: async () => {
+      return new DataLoader(
+        (ids: readonly string[]) => batchFetchCoingeckoPrices(ids, currency),
+        {
+          // workaround to work on Vercel Edge runtime
+          batchScheduleFn: (cb) => setTimeout(cb, 0),
+        }
+      );
```

Comment:
I wonder if we should create an `EdgeDataLoader` so that we don't have to continuously assign the `batchScheduleFn`.